### PR TITLE
Apply view model to GNSS receiver component

### PR DESCRIPTION
**Pull request by frankfu03**

### CODE REVIEW

File: src/client/app/gnss-receiver/gnss-receiver-group.component.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    super();
+  }
+
+  getItemName(): string {
+    return 'GNSS Receiver';
+  }
+
+  compare(obj1: GnssReceiverViewModel, obj2: GnssReceiverViewModel): number {
+    if (obj1 === null || obj2 === null) {
+      return 0;
```
Comment (lbodor) [inline src/client/app/gnss-receiver/gnss-receiver-group.component.ts]: Doesn't zero mean the objects are equal? Function compareDates() below returns -1, 0, or 1, depending on whether the answer is respectively less than, equal, or greater than.

Comment (frankfu03) [inline src/client/app/gnss-receiver/gnss-receiver-group.component.ts]: Yes, zero means the two objects are equal. The compare() method is used to sort a list of items in a descending date order. In this case, if either of the objects is null, there will be no sorting for them.

Comment (frankfu03) [inline src/client/app/gnss-receiver/gnss-receiver-group.component.ts]: Yes, the returned value zero means the two objects are equal. The compare() method is used to sort a list of items in a descending date order. In this case, if either of the objects is null, there will be no change in order for them.

**Comment by lbodor** [comment]
Rebase to resolve conflicts.

File: src/client/app/gnss-receiver/gnss-receiver-group.component.ts (hunk 1 above)
Comment (lbodor) [inline src/client/app/gnss-receiver/gnss-receiver-group.component.ts]: OK, for now. I've added a todo in humidity sensors to fix this. I don't think it's right for compare to return zero when its arguments are not equal. We will try to implement correct sorting and a correct compare function.

Comment (oehm-smith) [inline src/client/app/gnss-receiver/gnss-receiver-group.component.ts]: I made the change to remove these lines in HumiditySensors and have been using it for a couple days without any errors (related to this).  

You might as well remove these lines.

Comment (lbodor) [inline src/client/app/gnss-receiver/gnss-receiver-group.component.ts]: That's great news, let's remove them. I think we should turn on strict --strictNullChecks. See: https://blog.mariusschulz.com/2016/09/27/typescript-2-0-non-nullable-types.

**Pull request closed without merging**